lib/spack/spack/tag.py

```diff
@@ -112,14 +113,15 @@ def merge(self, other):
             spkgs, opkgs = self.tags[tag], other.tags[tag]
             self.tags[tag] = sorted(list(set(spkgs + opkgs)))
 
-    def update_package(self, pkg_name):
+    def update_package(self, pkg_name, repository=None):
         """Updates a package in the tag index.
 
         Args:
             pkg_name (str): name of the package to be removed from the index
 
         """
-        pkg_cls = spack.repo.path.get_pkg_class(pkg_name)
+        repository = repository or spack.repo.path
```

Comment:
Why can't this (and all other instance methods in the indexers) use `self.repository`?  It seems way cleaner to only pass that in the constructor than to require it on every method.

Reply:
This is a leftover, thanks for catching it. Did you seen any other on instance methods?